### PR TITLE
Allow DetGadget Hat and Lawbringer to work regardless of stammering and common mutantrace speeches

### LINKED ISSUE #7361: Mutantrace Lisps and the DetGadget Hat
### Labels

[BUG - TRIVIAL]

### Description

Half of the functions of the DetGadget hat do not work for mutantraces with lisps. Y'know, same issue the Lawbringer had.

### Steps to reproduce

1. Spawn in as a Saurian Detective.
2. Put your Det.net goggles into the DetGadget hat.
3. Say "go go gadget glassssssesss" a couple times.
4. Cry because you can't get the goggles out of the hat.

### Expected Behavior

I expected the hat to register "glassssssesss" as "glasses", so it would give me the glasses I put into the hat.

### Additional Information & Screenshots

Affected equipment:
Saurians - "Spray", "Flashlight", "Scanner", "Glasses"
Bovines - "Monitor"

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME OBJECTS][QoL]

## About the PR 
This PR allows the DetGadget Hat and Lawbringer to work even if you are stammering, and when playing with a lizard or cow mutantrace.

## Why's this needed? 
Would be cool to be able to any phrase when playing with a different mutantrace.

Also would be nice to use the Lawbringer when you are stammering

Fixes #7361

## Changelog
```changelog
(u)FlameArrow57
(+)The DetGadget Hat and Lawbringer are no longer affected by stammering or lizard and cow speech.
```

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
The alternate phrases already exist to allow mutantrace support and also “poisoning” the hos with booze to reduce there lawbringer effectiveness is a very funny and good strategy that would be a shame to be gone. This pr has basically no benefit, and its detriment is getting rid of a very entertaining aspect of these items.

**Comment by FlameArrow57** [comment]
The PR doesn't change the effects of ethanol with the Lawbringer.

**Comment by ThatFiveGuys** [comment]
> The PR doesn't change the effects of ethanol with the Lawbringer.

Still runs into same issue with stammering, also having it seemingly arbitrarily work for some voice changing things and not work for other voice changes seems really unintuitive

File: code/obj/item/gun/energy.dm [hunk 1]
```diff
@@ -1545,7 +1548,14 @@
...
 		msg = r.Replace(msg, "")
-		return msg
+		if (!msg)
+			return
+		var/new_message
+		for (var/i = 1 to length(msg) - 1)
+			if (msg[i] != msg[i + 1])
+				new_message += msg[i]
+		new_message += msg[length(msg)]
+		return new_message
```
Comment (ZeWaka) [inline code/obj/item/gun/energy.dm:1558]: what are you even doing here

File: code/obj/item/gun/energy.dm [hunk 2]
```diff
@@ -1463,13 +1468,11 @@
 
 					/datum/projectile/energy_bolt/pulse
 		else		//if you're not the owner and try to change it, then fuck you
-			switch(text)
-				if ("detain","execute","knockout","hotshot","incendiary","bigshot","highexplosive","he","clownshot","clown", "pulse", "punch")
```
Comment (ZeWaka) [inline code/obj/item/gun/energy.dm:1467]: ?

File: code/obj/item/gun/energy.dm [hunk 3]
```diff
@@ -1435,7 +1440,7 @@
 					current_projectile.cost = 50
 					item_state = "lawg-smokeshot"
 					playsound(M, 'sound/vox/smoke.ogg', 50)
-				if ("knockout", "sleepshot")
+				if ("knockout", "slepshot") // sleepshot -> slepshot
```
Comment (ZeWaka) [inline code/obj/item/gun/energy.dm:1443]: this isn't documented!

File: code/obj/item/clothing/hats.dm [hunk 4]
```diff
@@ -440,6 +444,15 @@ proc/filter_trait_hats(var/type)
...
 
+	proc/remove_repeated_letters(message)
+		var/msg = message
+		var/new_message
+		for (var/i = 1 to length(msg) - 1)
+			if (msg[i] != msg[i + 1])
+				new_message += msg[i]
+		new_message += msg[length(msg)]
+		return new_message
+
```
Comment (ZeWaka) [inline code/obj/item/clothing/hats.dm:455]: no, fix this another way

File: code/obj/item/clothing/hats.dm [hunk 5]
```diff
@@ -311,26 +311,26 @@ proc/filter_trait_hats(var/type)
...
-									"glasses" = /obj/item/clothing/glasses)
+		items = list("bodybag" = list("bodybag", /obj/item/body_bag),
+									"scaner" = list("scanner", /obj/item/device/detective_scanner), // scanner -> scaner for phrase detection
+									"lighter" = list("lighter", /obj/item/device/light/zippo/),
+									"spray" = list("spray", /obj/item/spraybottle),
+									"monitor" = list("monitor", /obj/item/device/camera_viewer),
+									"camera" = list("camera", /obj/item/camera),
+									"audiolog" = list("audiolog", /obj/item/device/audio_log),
+									"flashlight" = list("flashlight", /obj/item/device/light/flashlight),
+									"glases" = list("glasses", /obj/item/clothing/glasses)) // glasses -> glases
```
Comment (ZeWaka) [inline code/obj/item/clothing/hats.dm:322]: no

**Review by ZeWaka — CHANGES_REQUESTED** [review]
this is implemented in a completely wrong way, not mergeable whatsoever in its current approach

**Comment by FlameArrow57** [comment]
Anything in mind for this? I wasn't sure if you were meaning some way to do it without allowing less letter than intended, or to just rework what was passed into speak_into()

**Comment by ZeWaka** [comment]
> Anything in mind for this? I wasn't sure if you were meaning some way to do it without allowing less letter than intended, or to just rework what was passed into speak_into()

the speech pipeline should just be reworked entirely, so both the original and text-to-display pipeline can be modified/read by speech accents and such, this is a hacky bodge ontop of old smelly code

**Pull request closed without merging**